src/compiler/utilities.ts

```diff
@@ -1193,108 +1193,407 @@ export function getInternalEmitFlags(node: Node): InternalEmitFlags {
-            AsyncIterableIterator: emptyArray,
-            AsyncGenerator: emptyArray,
-            AsyncGeneratorFunction: emptyArray,
-            NumberFormat: ["formatToParts"]
-        },
-        es2019: {
-            Array: ["flat", "flatMap"],
-            ObjectConstructor: ["fromEntries"],
-            String: ["trimStart", "trimEnd", "trimLeft", "trimRight"],
-            Symbol: ["description"]
-        },
-        es2020: {
-            BigInt: emptyArray,
-            BigInt64Array: emptyArray,
-            BigUint64Array: emptyArray,
-            PromiseConstructor: ["allSettled"],
-            SymbolConstructor: ["matchAll"],
-            String: ["matchAll"],
-            DataView: ["setBigInt64", "setBigUint64", "getBigInt64", "getBigUint64"],
-            RelativeTimeFormat: ["format", "formatToParts", "resolvedOptions"]
-        },
-        es2021: {
-            PromiseConstructor: ["any"],
-            String: ["replaceAll"]
-        },
-        es2022: {
-            Array: ["at"],
-            String: ["at"],
-            Int8Array: ["at"],
-            Uint8Array: ["at"],
-            Uint8ClampedArray: ["at"],
-            Int16Array: ["at"],
-            Uint16Array: ["at"],
-            Int32Array: ["at"],
-            Uint32Array: ["at"],
-            Float32Array: ["at"],
-            Float64Array: ["at"],
-            BigInt64Array: ["at"],
-            BigUint64Array: ["at"],
-            ObjectConstructor: ["hasOwn"],
-            Error: ["cause"]
-        },
-        es2023: {
-            Array: ["findLastIndex", "findLast"],
-            Int8Array: ["findLastIndex", "findLast"],
-            Uint8Array: ["findLastIndex", "findLast"],
-            Uint8ClampedArray: ["findLastIndex", "findLast"],
-            Int16Array: ["findLastIndex", "findLast"],
-            Uint16Array: ["findLastIndex", "findLast"],
-            Int32Array: ["findLastIndex", "findLast"],
-            Uint32Array: ["findLastIndex", "findLast"],
-            Float32Array: ["findLastIndex", "findLast"],
-            Float64Array: ["findLastIndex", "findLast"],
-            BigInt64Array: ["findLastIndex", "findLast"],
-            BigUint64Array: ["findLastIndex", "findLast"],
-        }
-    };
+    return new Map(Object.entries({
+        Array: new Map(Object.entries({
+            es2015: [
+                "find",
+                "findIndex",
+                "fill",
+                "copyWithin",
+                "entries",
+                "keys",
+                "values"
+            ],
+            es2016: [
+                "includes"
+            ],
+            es2019: [
+                "flat",
+                "flatMap"
+            ],
+            es2022: [
+                "at"
+            ],
+            es2023: [
+                "findLastIndex",
+                "findLast"
+            ],
+        })),
+        Iterator: new Map(Object.entries({
+            es2015: emptyArray,

```

Comment:
this dual use is stil pretty confusing. 
I don't think I'd prefer a two-level map `Type -> property -> lib` and a separate one-level map `Type -> lib`, because it would involve more typing, but it would definitely be easier to understand.